src/Service/BucketFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Service;
+
+use Aws\S3\S3Client;
+use League\Flysystem\Filesystem;
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\Adapter\Local;
+use Symfony\Component\Filesystem\Exception\IOExceptionInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+
+class BucketFactory {
+
+    private $params;
+
+    public function __construct(ParameterBagInterface $params)
+    {
+        $this->params = $params;
+    }
+
+    public function filesystem () {
+        $adapter = new Local('/tmp/attachments');
+
+        if ($this->params->get('bucket_provider') == 'S3') {
+            $client = new S3Client([
```

Comment:
Probably should allow for other options to be passed as well, such as `endpoint` and `use_path_style_endpoint` (to be able to test with S3-compatible APIs such as Minio).